File: src/main/java/fr/vlegall/sochief/service/difficulty/RecipeDifficultyService.java
package fr.vlegall.sochief.service.difficulty;

import fr.vlegall.sochief.contracts.common.NamedIdDto;
import fr.vlegall.sochief.repository.RecipeDifficultyRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class RecipeDifficultyService implements IRecipeDifficultyService {
    private final RecipeDifficultyRepository difficultyRepository;
    private final RecipeDifficultyMapper recipeDifficultyMapper;

    public RecipeDifficultyService(RecipeDifficultyRepository difficultyRepository, RecipeDifficultyMapper recipeDifficultyMapper) {
        this.difficultyRepository = difficultyRepository;
        this.recipeDifficultyMapper = recipeDifficultyMapper;
    }

    @Override
    public List<NamedIdDto> getRecipeDifficulties() {
        return difficultyRepository.findAllByOrderByIdAsc()
                .stream()
                .map(recipeDifficultyMapper::toDto)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/vlegall/sochief/service/difficulty/RecipeDifficultyService.kt
package com.vlegall.sochief.service.difficulty

import com.vlegall.sochief.repository.RecipeDifficultyRepository
import com.vlegall.sochiefcontracts.dto.common.NamedIdDto
import org.springframework.stereotype.Service

@Service
class RecipeDifficultyService(
    private val difficultyRepository: RecipeDifficultyRepository,
    private val recipeDifficultyMapper: RecipeDifficultyMapper
) : IRecipeDifficultyService {
    override fun getRecipeDifficulties(): List<NamedIdDto> {
        return difficultyRepository.findAllByOrderByIdAsc().map(recipeDifficultyMapper::toDto)
    }

}